cob_scan_unifier/src/scan_unifier_node.cpp

```diff
@@ -123,150 +166,99 @@ void scan_unifier_node::getParams()
+
+  sensor_msgs::LaserScan unified_scan = unifyLaserScans(current_scans);
+
+  ROS_DEBUG("Publishing unified scan.");
+  topicPub_LaserUnified_.publish(unified_scan);
 }
 
 /**
- * @function unifieLaserScans
+ * @function unifyLaserScans
  * @brief unifie the scan information from all laser scans in vec_laser_struct_
  *
  * input: -
  * output:
  * @param: a laser scan message containing unified information from all scanners
  */
-sensor_msgs::LaserScan scan_unifier_node::unifieLaserScans()
+sensor_msgs::LaserScan ScanUnifierNode::unifyLaserScans(std::vector<sensor_msgs::LaserScan::ConstPtr> current_scans)
 {
   sensor_msgs::LaserScan unified_scan = sensor_msgs::LaserScan();
   std::vector<sensor_msgs::PointCloud> vec_cloud;
   vec_cloud.assign(config_.number_input_scans, sensor_msgs::PointCloud());
 
-  if(!vec_laser_struct_.empty())
+  if(!current_scans.empty())
   {
     ROS_DEBUG("start converting");
     for(int i=0; i < config_.number_input_scans; i++)
     {
-      vec_cloud.at(i).header.stamp = vec_laser_struct_.at(i).current_scan_msg.header.stamp;
-      ROS_DEBUG_STREAM("Converting scans to point clouds at index: " << i << ", at time: " << vec_laser_struct_.at(i).current_scan_msg.header.stamp << " now: " << ros::Time::now());
+      vec_cloud.at(i).header.stamp = current_scans.at(i)->header.stamp;
+      ROS_DEBUG_STREAM("Converting scans to point clouds at index: " << i << ", at time: " << current_scans.at(i)->header.stamp << " now: " << ros::Time::now());
       try
       {
-        listener_.waitForTransform("/base_link", vec_laser_struct_.at(i).current_scan_msg.header.frame_id,
-            vec_laser_struct_.at(i).current_scan_msg.header.stamp, ros::Duration(3.0));
+        if (!listener_.waitForTransform(frame_, current_scans.at(i)->header.frame_id,
+                                        current_scans.at(i)->header.stamp, ros::Duration(3.0)))
+        {
+          continue;
+        }
+        else
+        {
+          ROS_WARN_STREAM("Scan unifier skipped scan with " << current_scans.at(i)->header.stamp << " stamp.");
+        }
 
         ROS_DEBUG("now project to point_cloud");
-        projector_.transformLaserScanToPointCloud("/base_link",vec_laser_struct_.at(i).current_scan_msg, vec_cloud.at(i), listener_);
+        projector_.transformLaserScanToPointCloud(frame_,*current_scans.at(i), vec_cloud.at(i), listener_);
       }
-      catch(tf::TransformException ex){
+      catch(tf::TransformException &ex){
         ROS_ERROR("%s",ex.what());
       }
     }
     ROS_DEBUG("Creating message header");
-    unified_scan.header = vec_laser_struct_.at(0).current_scan_msg.header;
-    unified_scan.header.frame_id = "base_link";
+    unified_scan.header = current_scans.at(0)->header;
+    unified_scan.header.frame_id = frame_;
     unified_scan.angle_increment = M_PI/180.0/2.0;
     unified_scan.angle_min = -M_PI + unified_scan.angle_increment*0.01;
     unified_scan.angle_max =  M_PI - unified_scan.angle_increment*0.01;
```

Comment:
This might actually be the reason for #249 

Also, the above restricts this unifier to be used with a spacing fo 0.5°. This should probably be adapted as well.
Something like "biggest common denominator" but in float ;-)
Added it to #249